pkg/util/log/log.go

```diff
@@ -33,7 +35,11 @@ func InitLogger(cfg *server.Config) {
 	// this works in go 1.12, but doesn't work in versions earlier.
 	// it will always shows the wrapper function generated by compiler
 	// marked <autogenerated> in old versions.
-	cfg.Log = logging.GoKit(log.With(l, "caller", log.Caller(4)))
+	srvLog := log.With(l, "caller", log.Caller(4))
+	dedups = append(dedups, NewDedupLogger("push.go", []string{"level", "user", "err"}, srvLog))
```

Comment:
I don't think deduping on `err` will work, because error messages are different for almost every single error (e.g. limit errors). I think could be more interesting deduplicating on the new global error ID (when available).